### PR TITLE
use hooks

### PR DESCRIPTION
**Pull request by msuntharesan**
Using it
```js
import { Provider, useUnstated } from '../src/unstated';

class CounterContainer extends Container {
  state = { count: 0 };

  increment() {
    this.setState({ count: this.state.count + 1 });
  }

  decrement() {
    this.setState({ count: this.state.count - 1 });
  }
}


function Counter() {
  const [counter] = useUnstated(CounterContainer);
  return (
    <div>
      <button onClick={() => counter.decrement()}>-</button>
      <span>{counter.state.count}</span>
      <button onClick={() => counter.increment()}>+</button>
    </div>
  );
}


render(
  <Provider>
    <Counter />
    <CounterUsingHook />
  </Provider>,
  document. getElementById('root')
);

```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "create-react-context": "^0.1.5"
```
Comment (SamyPesse) [inline package.json]: This dependency can be removed

**Review by SamyPesse — CHANGES_REQUESTED** [review]
Thank you @vanthiyathevan 👏 

I've published this PR under a fork: https://github.com/GitbookIO/unstated/commit/083a1ab139c806e0bf1a3802fb2d6efd71d986c8 that can be used as a git dependency.

**Comment by vanthiyathevan** [comment]
@SamyPesse Thank you. I've switched using `useRef` for instances. Also switched `Provider` to use the hooks.

**Comment by jamiebuilds** [comment]
I'm questioning the name `useUnstated`, I think I would prefer `useContainer`

I also think it would be preferable to `use` one container at a time:

```js
let foo = useContainer(Foo)
let bar = useContainer(Bar)
```

**Comment by zaguiini** [comment]
Also, what if I want to subscribe to multiple Containers? The argument to `useContainer` (as pointed by Jamie) should be an array of containers instead of one single... Does that make sense to you?

**Comment by jamiebuilds** [comment]
I only added the array syntax to Unstated because nesting render functions was really annoying. But I don't think the problem exists in hooks, so I'd rather keep the API as plain as possible

**Comment by Airboy** [comment]
Thank you @vanthiyathevan. That's a very interesting proposal. Any update on this?

**Comment by artemkloko** [comment]
I tried this on a react-native app that already uses `unstated` with success and it did not work. I did not switch to `useContainer` in my try, I tried it with the already available `<Subscribe />`that were in the app. 

Although I am only getting a warning, the state is not being propagated to the subscribers
```
Warning: State updates from the useState() and useReducer() Hooks don't support the second callback argument. To execute a side effect after rendering, declare it in the component body with useEffect().
```

Here is my branch based on @msuntharesan with the fixes for previously mentioned things and some other minor stuff 
https://github.com/artemkloko/unstated/tree/useContainer

If anyone wants to try it out, I also have a branch with built `lib` 
https://github.com/artemkloko/unstated/tree/useContainer_merged

And with promises/callback removed, as I though it was the problem that the warning was referring to 
https://github.com/artemkloko/unstated/tree/useContainer_no-promises_merged

**Pull request closed without merging**